<a href="https://colab.research.google.com/github/ericnsoares/PUCBIMaster/blob/main/Iris.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Análise de Previsão de Churn


Neste script iremos prever a espécie de uma planta usando o algoritmo KNN (K Nearest Neighbor).

# Setup

##Importar bibliotecas necessárias

In [1]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.neighbors import KNeighborsClassifier
from sklearn.metrics import accuracy_score

# Exercício 1

Carregue o arquivo csv chamado 'Iris.csv' usando pandas e salve a base em uma variável chamada df. (dica: pd.read_csv('<nome_da_base.csv>')) Imprima as primeiras 10 linhas do dataframe e verifique se tudo foi carregado corretamente. (dica: df.head())

In [2]:
df = pd.read_csv("_Iris.csv")

In [3]:
print(df.head(10))


   SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm      Species
0            5.1           3.5            1.4           0.2  Iris-setosa
1            4.9           3.0            1.4           0.2  Iris-setosa
2            4.7           3.2            1.3           0.2  Iris-setosa
3            4.6           3.1            1.5           0.2  Iris-setosa
4            5.0           3.6            1.4           0.2  Iris-setosa
5            5.4           3.9            1.7           0.4  Iris-setosa
6            4.6           3.4            1.4           0.3  Iris-setosa
7            5.0           3.4            1.5           0.2  Iris-setosa
8            4.4           2.9            1.4           0.2  Iris-setosa
9            4.9           3.1            1.5           0.1  Iris-setosa


# Exercício 2: 

Que coluna indica a espécie da planta? Quantas plantas existem na sua base de dados? (dica: len(?))


In [11]:
#coluna rótulo: <Species>
print(df.Species)
# número de plantas:
print (len(df))


0         Iris-setosa
1         Iris-setosa
2         Iris-setosa
3         Iris-setosa
4         Iris-setosa
            ...      
145    Iris-virginica
146    Iris-virginica
147    Iris-virginica
148    Iris-virginica
149    Iris-virginica
Name: Species, Length: 150, dtype: object
150


# Exercício 3:

Use df.describe() para explorar cada coluna. Todos devem ser igual a 150 pois não temos missing values!!

In [6]:
df.describe()

SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm
count     150.000000    150.000000     150.000000    150.000000
mean        5.843333      3.054000       3.758667      1.198667
std         0.828066      0.433594       1.764420      0.763161
min         4.300000      2.000000       1.000000      0.100000
25%         5.100000      2.800000       1.600000      0.300000
50%         5.800000      3.000000       4.350000      1.300000
75%         6.400000      3.300000       5.100000      1.800000
max         7.900000      4.400000       6.900000      2.500000

# Exercício 5:

Verifique o número de colunas da base de dados (dica: df.shape)

Separe a base de dados em atributos de entrada e rótulos. Associe os atributos a variável X e os rótulos a variável y.

Verifique as primeiras linhas das bases separadas.

In [15]:
print(df.shape) #Retorna uma tupla q representa a dimensionalidade (linhas X Colunas) do DataFrame
print(df.shape[1]) #A quantidade de colunas é o segundo valor da tupla

(150, 5)
5


In [28]:
x = df.iloc[:, 0:4]
print ("x:\n", x)
y = df.loc[:, 'Species']
print ("y:\n", y)

x:
      SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm
0              5.1           3.5            1.4           0.2
1              4.9           3.0            1.4           0.2
2              4.7           3.2            1.3           0.2
3              4.6           3.1            1.5           0.2
4              5.0           3.6            1.4           0.2
..             ...           ...            ...           ...
145            6.7           3.0            5.2           2.3
146            6.3           2.5            5.0           1.9
147            6.5           3.0            5.2           2.0
148            6.2           3.4            5.4           2.3
149            5.9           3.0            5.1           1.8

[150 rows x 4 columns]
y:
 0         Iris-setosa
1         Iris-setosa
2         Iris-setosa
3         Iris-setosa
4         Iris-setosa
            ...      
145    Iris-virginica
146    Iris-virginica
147    Iris-virginica
148    Iris-virginica
149 

In [29]:
# primeiras linhas de X
print(x.head())

   SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm
0            5.1           3.5            1.4           0.2
1            4.9           3.0            1.4           0.2
2            4.7           3.2            1.3           0.2
3            4.6           3.1            1.5           0.2
4            5.0           3.6            1.4           0.2


In [31]:
# primeiras linhas de y
print(y.head())

0    Iris-setosa
1    Iris-setosa
2    Iris-setosa
3    Iris-setosa
4    Iris-setosa
Name: Species, dtype: object


# Exercício 6: 

Separa a base de treino (70%) e teste (30%) e associe-as as variáveis X_train, X_test, y_train, y_test (dica: train_test_split(X,y, test_size=0.3))

# Exercício 7: 

Crie um classificador KNN com 5 vizinhos. (hint: KNeighborsClassifier(n_neighbors=5))

In [ ]:
knn = 

# Exercício 8:

Apresente a base de treino para ajustar o modelo. (hint: knn.fit(<'base_de_treino_atributos'>, <'base_de_treino_rotulos'>))

In [ ]:
knn = 

# Exercício 9:

Use o modelo para fazer previsões na base de teste. Associe os resultados de previsão a variável y_pred. Dica: knn.predict(<'atributos de teste'>)

In [ ]:
y_pred = 

# Exercício 10:

Determine quão acurado é o seu modelo. (dica: accuracy_score(<'rotulo_real'>, <'rotulo_previsto'>))